### ISSUE TITLE
map.setStyle() causes TypeError: Cannot read property 'transition' of undefined

### ISSUE THREAD

**Issue by jmking** [mapbox/mapbox-gl-js#6043]
**mapbox-gl-js version: **0.43.0

After updating to 0.43.0, I see a console error whenever changing map the style via map.setStyle(). I did not see this in any previous versions. The error is:

```
mapbox-gl.js:520 Uncaught TypeError: Cannot read property 'transition' of undefined 
  at e._render (mapbox-gl.js:520)  
  at mapbox-gl.js:520
```

The map continues to function normally after the console error.

### Steps to Trigger Behavior

 1. Update from version 0.42.2 to 0.43.0
 2. Change map style using map.setStyle()

So far I've been unable to reproduce this in a JSFiddle, only within the application I'm working on. I'll continue trying to reproduce if necessary.

**Comment by lbud** [comment]
@jmking thanks for using Mapbox and sorry to hear you're running into an issue. We haven't seen this error on our end and I'm unable to reproduce it. Are you able to reproduce this in your application on the mapbox-gl-js master branch? You can grab a built copy [here](https://gist.githubusercontent.com/lbud/2713ad602215a71004172a44efed10c4/raw/52c107c713bcba228ffd1b9fc83d03b1e7c2df4f/mapbox-gl.js).

If the issue persists, we'll need a JSFiddle to diagnose it; let us know if you're able to create a minimal repro.

**Comment by lbud** [comment]
I'm still not able to reproduce with those instructions; we'll need a self-contained JSFiddle or the like to diagnose it.

**Comment by wxqqq** [comment]
https://jsfiddle.net/ygyu8fqn/

**Comment by ansis** [comment]
Thanks for the example @wxqqq! It looks like this has been fixed by https://github.com/mapbox/mapbox-gl-js/commit/d957a0564524b85df0410808d5dd476f2a2b5fe6 which will be included in v0.44.0, which is set to be released in the next couple days.
